File: 05 SWEA/D2/Java/src/Prob1926.java

/**
* title  : 1926. 간단한 369게임 (D2)
* time   : 104ms
* memory : 18592kb
*/

import java.io.*;
 
public class Prob1926 {
    
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int N = Integer.parseInt(br.readLine());
        for (int i = 1; i <= N; i++) {
            sb.append(checkNumber(String.valueOf(i))).append(" ");
        }

        System.out.println(sb);
    }

    public static String checkNumber(String number) {

        int count = 0;
        for (int i = 0; i < number.length(); i++) {
            int check = Character.getNumericValue(number.charAt(i));
            if (check > 0 && check % 3 == 0) {
                count++;
            }
        }

        return (count == 0) ? number : printNumber(count);
    }

    public static String printNumber(int count) {

        String str = "";
        for (int i = 0; i < count; i++) {
            str += "-";
        }

        return str;
    }
}

File: 05 SWEA/D2/Python/Prob1926.py

"""
title  : 1926. 간단한 369게임 (D2)
time   : 133ms
memory : 44792kb
"""

def solution():

    N = int(input())

    for num in range(1, N+1):
        print(count3(str(num)), end=" ")

def count3(num:str):

    count = 0
    for n in num:
        nnum = int(n)
        if (nnum > 0 and nnum % 3 == 0):
            count += 1

    return "-" * count if count else num

solution()